cartography/intel/aws/__init__.py

```diff
@@ -1,35 +1,78 @@
 import logging
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import List
 
 import boto3
 import botocore.exceptions
+import neo4j
 
-from . import dynamodb
 from . import ec2
-from . import ecr
-from . import eks
-from . import elasticsearch
-from . import emr
-from . import iam
-from . import kms
-from . import lambda_function
 from . import organizations
-from . import permission_relationships
-from . import rds
-from . import redshift
-from . import resourcegroupstaggingapi
-from . import route53
-from . import s3
+from .resources import RESOURCE_FUNCTIONS
+from cartography.intel.aws.util.common import parse_and_validate_aws_requested_syncs
 from cartography.util import run_analysis_job
 from cartography.util import run_cleanup_job
 from cartography.util import timeit
 
 logger = logging.getLogger(__name__)
 
 
-def _sync_one_account(neo4j_session, boto3_session, account_id, sync_tag, common_job_parameters):
-    iam.sync(neo4j_session, boto3_session, account_id, sync_tag, common_job_parameters)
-    s3.sync(neo4j_session, boto3_session, account_id, sync_tag, common_job_parameters)
+def _build_aws_sync_kwargs(
+    neo4j_session: neo4j.Session, boto3_session: boto3.session.Session, regions: List[str], current_aws_account_id: str,
+    sync_tag: int, common_job_parameters: Dict[str, Any],
+) -> Dict[str, Any]:
+    return {
+        'neo4j_session': neo4j_session,
+        'boto3_session': boto3_session,
+        'regions': regions,
+        'current_aws_account_id': current_aws_account_id,
+        'update_tag': sync_tag,
+        'common_job_parameters': common_job_parameters,
+    }
+
+
+def _sync_one_account(
+    neo4j_session: neo4j.Session,
+    boto3_session: boto3.session.Session,
+    current_aws_account_id: str,
+    update_tag: int,
+    common_job_parameters: Dict[str, Any],
+    regions: List[str] = [],
```

Comment:
This is a new param! Future work/other custom scripts can now feed in specific regions if they want to.